picard/ui/searchdialog.py

```diff
@@ -0,0 +1,403 @@
+        self.adv_opt_row_widget.setLayout(self.adv_opt_row_layout)
+        self.layout.addWidget(self.adv_opt_row_widget)
+        self.layout.setContentsMargins(1, 1, 1, 1)
+        self.layout.setSpacing(1)
+        self.setMaximumHeight(60)
+
+    def search(self):
+        self.parent.search(self.search_edit.text())
+
+    def restore_checkbox_state(self):
+        self.use_adv_search_syntax.setChecked(config.setting["use_adv_search_syntax"])
+
+    def save_checkbox_state(self):
+        config.setting["use_adv_search_syntax"] = self.use_adv_search_syntax.isChecked()
+
+
+class SearchDialog(PicardDialog):
+
+    options = [
+        config.Option("persist", "searchdialog_window_size", QtCore.QSize(720, 360)),
+        config.Option("persist", "searchdialog_header_state", QtCore.QByteArray())
+    ]
+
+    def __init__(self, parent=None):
+        PicardDialog.__init__(self, parent)
+        self.search_results = []
+        self.setupUi()
+        self.restore_state()
+
+    def setupUi(self):
+        self.verticalLayout = QtGui.QVBoxLayout(self)
+        self.verticalLayout.setObjectName(_("vertical_layout"))
+        self.search_box = SearchBox(self)
+        self.search_box.setObjectName(_("search_box"))
+        self.verticalLayout.addWidget(self.search_box)
+        self.center_widget = QtGui.QWidget(self)
+        self.center_widget.setObjectName(_("center_widget"))
+        self.center_layout = QtGui.QVBoxLayout(self.center_widget)
+        self.center_layout.setObjectName(_("center_layout"))
+        self.center_layout.setContentsMargins(1, 1, 1, 1)
+        self.center_widget.setLayout(self.center_layout)
+        self.verticalLayout.addWidget(self.center_widget)
+        self.buttonBox = QtGui.QDialogButtonBox(self)
+        self.buttonBox.setObjectName(_("button_box"))
+        self.load_button = QtGui.QPushButton(_("Load into Picard"))
+        self.load_button.setEnabled(False)
+        self.buttonBox.addButton(
+                self.load_button,
+                QtGui.QDialogButtonBox.AcceptRole)
+        self.buttonBox.addButton(
+                StandardButton(StandardButton.CANCEL),
+                QtGui.QDialogButtonBox.RejectRole)
+        self.buttonBox.accepted.connect(self.accept)
+        self.buttonBox.rejected.connect(self.reject)
+        self.verticalLayout.addWidget(self.buttonBox)
+
+    def add_widget_to_center_layout(self, widget):
+        wid = self.center_layout.itemAt(0)
+        if wid:
+            if wid.widget().objectName() == "results_table":
+                self.table = None
+            wid.widget().deleteLater()
+        self.center_layout.addWidget(widget)
+
+    def show_progress(self):
+        self.progress_widget = QtGui.QWidget(self)
+        self.progress_widget.setObjectName("progress_widget")
+        layout = QtGui.QVBoxLayout(self.progress_widget)
+        text_label = QtGui.QLabel('<strong>Fetching results...</strong>', self.progress_widget)
```

Comment:
That needs to be translated. Also, we already use `_('Loading...')` elsewhere, so maybe it'd be better to reuse the existing translation.